graylog2-server/src/main/java/org/graylog2/lookup/adapters/CSVFileDataAdapter.java

```diff
@@ -245,6 +252,27 @@ public LookupResult doGet(Object key) {
         return LookupResult.single(value);
     }
 
+    public LookupResult getResultForCIDRRange(Object ip) {
+        for (Map.Entry<String, String> entry : lookupRef.get().entrySet()) {
+            String range = entry.getKey();
+            Optional<IpSubnet> subnet = ReservedIpChecker.stringToSubnet(range);
+            if (subnet.isEmpty()) {
+                LOG.warn("CIDR range '{}' in data adapter '{}' is not a valid subnet, skipping this key in lookup.", entry, name);
+            } else {
+                try {
+                    if (subnet.get().contains(String.valueOf(ip))) {
+                        return LookupResult.single(entry.getValue());
+                    }
+                } catch (UnknownHostException e) {
+                    LOG.warn("Attempted to do a CIDR range lookup on invalid IP '{}'", ip);
```

Comment:
We should consider changing the log level for both statements to DEBUG because WARN can result in **lots** of server log messages. :smile: Misconfigured lookups or unexpected field values can become a problem with high message rates.

Reply:
Good call! Have dropped level to debug in latest commit